models/gas/optimism/gas_optimism_fees.sql

```diff
@@ -0,0 +1,51 @@
+{{ config(
+    alias = 'fees',
```

Comment:
Another useful add could be:

`( length( replace( decode(unhex(substring(data,3)), 'US-ASCII') , chr(0), '')) )` AS num_nonzero_bytes
`( length( decode(unhex(substring(data,3)), 'US-ASCII') ) - length(replace( decode(unhex(substring(data,3)), 'US-ASCII') , chr(0), '')) )` AS num_zero_bytes

```
16 * ( length( replace( decode(unhex(substring(data,3)), 'US-ASCII') , chr(0), '')) ) --16 * nonzero bytes
                + 4 * ( length( decode(unhex(substring(data,3)), 'US-ASCII') ) - length(replace( decode(unhex(substring(data,3)), 'US-ASCII') , chr(0), '')) ) --4 * zero bytes
```
AS calldata_gas

like: https://dune.com/queries/876366/1573076

Adding this + the l1 tx fee equivalent would solve most custom code for Optimism gas calcs.